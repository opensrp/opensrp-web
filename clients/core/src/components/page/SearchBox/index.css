.search-bar-icon {
    position: absolute;
    color: #c2c2c2;
    padding: 8px 18px;
    z-index: 10;
    font-size: 12px;
}

.search-bar-input {
    padding-left: 56px;
    caret-color: #0192d4;
    font-size: 12px;
}

.search-bar-input::placeholder {
    color: #c2c2c2;
}

.form-control {
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
}

#custom-search-input {
    max-width: 200px;
}
