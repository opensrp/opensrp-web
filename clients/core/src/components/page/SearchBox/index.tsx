import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { library } from '@fortawesome/fontawesome-svg-core';
import * as React from 'react';
import { Input, InputGroup } from 'reactstrap';
import './index.css';
import { fas } from '@fortawesome/free-solid-svg-icons';

library.add(fas);

/** props interface for SearchBox */
export interface SearchBoxProps {
    placeholder?: string;
    searchCallBack(searchString: string): void;
}

/** state interface for SearchBox  */
export interface SearchBoxState {
    searchText: string;
}

const searchBoxdefaultProps: SearchBoxProps = {
    placeholder: 'Search',
    searchCallBack: searchText => searchText,
};

/** default state for searchBox */
const defaultSearchBoxState = {
    searchText: '',
};

/** SearchBox component */
class SearchBox extends React.Component<SearchBoxProps, SearchBoxState> {
    public static defaultProps: SearchBoxProps = searchBoxdefaultProps;
    constructor(props: SearchBoxProps) {
        super(props);
        this.state = defaultSearchBoxState;
    }
    public render() {
        return (
            <div id="custom-search-input">
                <InputGroup className="shadow-sm">
                    <span className="input-group-btn search-bar-icon">
                        <FontAwesomeIcon icon="search" />
                    </span>
                    <Input
                        placeholder={this.props.placeholder}
                        className="search-bar-input"
                        onChange={this.handleChange}
                        onKeyPress={this.search}
                    />
                </InputGroup>
            </div>
        );
    }
    /** change the searchText on change */
    private handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        this.setState({ searchText: event.target.value });
    };

    /** invoke the callback function to lifting up the state */
    private search = (event: React.KeyboardEvent<HTMLDivElement>) => {
        const { searchCallBack } = this.props;
        const { searchText } = this.state;
        if (event.key === 'Enter') {
            searchCallBack(searchText);
        }
    };
}

export default SearchBox;
