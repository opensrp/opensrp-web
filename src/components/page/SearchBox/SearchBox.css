.search-bar-icon {
  position: absolute;
  color: #C2C2C2;
  padding: 8px 18px;
  z-index: 10;
  font-size:12px;
}

.search-bar-input {
  padding-left: 56px;
  caret-color: #0192D4;
  font-size: 12px;
}

.search-bar-input::placeholder{
  color: #C2C2C2;
}

.form-control {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}
