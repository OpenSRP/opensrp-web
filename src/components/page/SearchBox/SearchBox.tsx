import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import * as React from 'react';
import { Input, InputGroup, InputGroupAddon, InputGroupText } from 'reactstrap';

/** state interface for SearchBox  */
interface SearchBoxState {
  searchText: '';
}

class SearchBox extends React.Component<{}, SearchBoxState> {
  public render() {
    return (
      <InputGroup>
        <InputGroupAddon addonType="prepend">
          <InputGroupText>
            <FontAwesomeIcon icon={['fas', 'search']} />
          </InputGroupText>
        </InputGroupAddon>
        <Input placeholder="search" />
      </InputGroup>
    );
  }
}

export default SearchBox;
