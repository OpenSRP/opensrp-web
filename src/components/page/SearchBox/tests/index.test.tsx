import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import SearchBox, { SearchBoxProps } from '..';

const callBackMock = jest.fn();
const props: SearchBoxProps = {
  placeholder: 'Search Household',
  searchCallBack: callBackMock,
};

describe('components/page/Searchbox', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    const component = shallow(<SearchBox {...props} />);
    const wrapper = component.find('#custom-search-input');
    expect(toJson(wrapper.find('.search-bar-input'))).toMatchSnapshot('SearchBox input');
    expect(toJson(wrapper.find('.search-bar-icon'))).toMatchSnapshot('SearchBox ion');
    expect(wrapper.length).toBe(1);
  });

  it('invokes callback correctly', () => {
    const wrapper = shallow(<SearchBox {...props} />);

    // simulate typing some text
    const inputWrapper = wrapper.find('.search-bar-input');
    expect(inputWrapper.length).toEqual(1);

    inputWrapper.simulate('change', { target: { value: 'randomString' } });

    // callback is not yet called;
    expect(callBackMock).not.toHaveBeenCalled();

    // simulate click on enter key
    inputWrapper.simulate('keypress', { key: 'Enter' });

    // now callback should have been called with "'randomString'"
    expect(callBackMock).toHaveBeenCalledTimes(1);
    expect(callBackMock).toHaveBeenCalledWith('randomString');
  });
});
