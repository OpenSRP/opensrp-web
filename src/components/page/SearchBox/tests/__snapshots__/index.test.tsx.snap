// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/page/Searchbox renders without crashing: SearchBox input 1`] = `
<Input
  className="search-bar-input"
  onChange={[Function]}
  onKeyPress={[Function]}
  placeholder="Search Household"
  type="text"
/>
`;

exports[`components/page/Searchbox renders without crashing: SearchBox ion 1`] = `
<span
  className="input-group-btn search-bar-icon"
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon={
      Array [
        "fas",
        "search",
      ]
    }
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
</span>
`;
